archCompare/config/compareMethods.json

```diff
@@ -0,0 +1,46 @@
+{
+  "_#comment": [
+    "Only allow fields listed, not _comment1/2/3 use an array of '_#comment'",
+    "This file shouldn't be included in the package unless trimmed to only basic unix commands"
+  ],
+  "preprocessors": {
+    "vcf.gz": {
+      "_#comment": [
+        "Preprocess both filea and fileb, then run the rule for 'vcf' extension",
+        "Extension provided as key is replaced by extension in 'then' when naming 'tmp'",
+        "This is just an example of a pre-processor, if zdiff works use it"
+      ],
+      "preprocess": "zcat {file} > {tmp}",
+      "then": "vcf"
+    }
+  },
+  "rules": {
+    "txt": {
+      "_#comment": [
+        "Use of standard diff command, fail when non-zero exit code"
+      ],
+      "cmd": "diff -q {filea} {fileb}",
+      "check": "exit-code"
+    },
+    "vcf": {
+      "_#comment": [
+        "Use of standard diff command which is expected to have differences",
+        "that are not a problem and can be filtered to give a final line count",
+        "of '0'"
+      ],
+      "cmd": "diff -y --suppress-common-lines {filea} {fileb} | grep -cvE '^##'",
```

Comment:
diff --suppress-common-lines -y  <( zcat {filea} | cut --complement -f 3  ) <( zcat {fileb} | cut --complement -f 3 ) |    grel -cvE '^##'  complement required to get rid of ID column otherwise rows will be never comparable

Reply:
Yes I'm aware of the compliment bit.  The point of this branch is to clarify that generating a correct diff is the responsibility of the user and config file, not the codebase.

Reply:
Filtering the relevant columns should be in the pre-process section. I've pushed a change with even more details.  FYI this was uploaded for Dave and I to discuss before asking for changes.

Reply:
See : https://github.com/cancerit/archCompare/blob/9f6d8a08803ce8e57bcb91f06280cbe16478c3db/archCompare/config/compareMethods.json

Reply:
FYI [this](https://github.com/cancerit/archCompare/blob/9f6d8a08803ce8e57bcb91f06280cbe16478c3db/archCompare/config/compareMethods.json#L11) comment line should have been prefixed `default:`